comp-mng-server/src/main/java/novi/uni/compserver/initialization/StartUpAppListener.java

```diff
@@ -0,0 +1,35 @@
+package novi.uni.compserver.initialization;
+
+import novi.uni.compserver.payload.responses.ApiResponse;
+import novi.uni.compserver.services.RegistrationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StartUpAppListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartUpAppListener.class);
+
+    @Autowired
+    RegistrationService registrationService;
+
+    /**
+     * Deze methode wordt bij elke opstart aangeroepen
+     * Echter zullen bij meerdere aanroepen de gebruikers niet aangemaakt omdat ze al bestaan
+     * Dit is omdat de database update aan staat in application.properties
+     * Maar het is nodig om initieel een 'gebruiker' aan te maken met ieder een andere rol
+     * Dit zorgt ervoor dat met beide rollen de functionaliteit van de app te gebruiken is
+     */
+    @EventListener
+    public void initializeEmpAndAdmin(ContextRefreshedEvent event) {
```

Comment:
Kun je denk ik beter gewoon in een SQL scriptje stoppen en je DB daarmee initialiseren, niet vanuit de java code

Reply:
Dat is niet erg, je slaat je sql file in zo een geval encrypted op in git (zie git-crypt)

Reply:
En in die file mag je gewoon het encrypted password zetten (eenmalig berekenen of via een stored precedure). Nu heb je hardcoded user en pass in je code staan. Misschien acceptabel voor dit project, maar daar krijg je denk ik wel een opmerking over.

Reply:
thanks, zal even loeren naar oplossingen